### PR TITLE
EVG-6595: Get all patches for a project

### PR DESCRIPTION
**Pull request by SupaJoon**
https://jira.mongodb.org/browse/EVG-6595

Get all of the patches for a project by calling getProjectPatches with the project name.
There are also code changes in spruce to complete this feature.

### CODE REVIEW

File: src/evergreen.ts [hunk 1]
```diff
@@ -118,6 +118,19 @@ export class client {
     return this.getUIResource(resource, params);
   }
 
+  /**
+   * Gets patches for a particular project
+   * @param projectName - Evergreen project name
+   * @returns a promise for the caller to handle responses
+   */
+  public getProjectPatches(projectName?: string, page?: number): AxiosPromise<models.Patches> {
+    const resource = "json/patches/project/" + projectName
```
Comment (tgrander) [inline src/evergreen.ts]: nit: prefer the use of template literals over string concatenation, i.e. `json/patches/project/${projectName}`

**Review by tgrander — CHANGES_REQUESTED** [review]

File: src/evergreen.ts [hunk 2]
```diff
@@ -118,6 +118,19 @@ export class client {
...
+    const params = {
+      page: page,
```
Comment (tgrander) [inline src/evergreen.ts]: don't have to include the value since the key and value are the same

**Review by tgrander — CHANGES_REQUESTED** [review]

**Review by john-m-liu — APPROVED** [review]
Looks good but be sure to increment the version number before merging (we'll also need to publish to npm, but we can do that later)

**Review by tgrander — APPROVED** [review]

**Comment by john-m-liu** [comment]
There's a mismatch between our linter and the default prettier settings, where one is checking for a trailing comma in the last field of a JSON object and the other isn't. Which way do we want to go with this? I personally like having the trailing comma since it makes adding new fields easier

**Comment by tgrander** [comment]
@john-m-liu i also like trailing comma

**Comment by john-m-liu** [comment]
Arjun if you agree, then let's add a prettier section in our package.json file with `trailingComma` set to `es5`

**Pull request merged**